projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/RepositoryRetryOperationsConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.aop.framework.Advised;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.Ordered;
+import org.springframework.data.repository.Repository;
+import org.springframework.lang.Nullable;
+import org.springframework.retry.interceptor.RetryInterceptorBuilder;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+
+@Slf4j
+public class RepositoryRetryOperationsConfiguration {
+
+   @Bean
+   public RepositoryRetryOperationsInterceptor repositoryRetryOperationsInterceptor() {
+      return new RepositoryRetryOperationsInterceptor();
+   }
+
+   class RepositoryRetryOperationsInterceptor implements BeanPostProcessor, Ordered {
+
+      @Override
+      public int getOrder() {
+         return Integer.MAX_VALUE;
+      }
+
+      @Override
+      @Nullable
+      public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+         if (bean instanceof Repository) {
+            Advised advised = (Advised) bean;
+            RetryOperationsInterceptor interceptor = RetryInterceptorBuilder.stateless()
```

Comment:
From [the documentation](https://docs.spring.io/spring-retry/docs/api/current/org/springframework/retry/interceptor/RetryOperationsInterceptor.html):

> If you want to retry a failed transaction you need to make sure that the transaction boundary is inside the retry, otherwise the successful attempt will roll back with the whole transaction. If the method being intercepted is also transactional, then use the ordering hints in the advice declarations to ensure that this one is before the transaction interceptor in the advice chain.

Have you considered this in the context of https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMonitor.java#L177?